osf/models/files.py

```diff
@@ -789,66 +484,141 @@ def serialize(self):
                 checkout=self.checkout._id if self.checkout else None,
             )
 
-        version = self.versions.last()
         return dict(
-            super(File, self).serialize(),
-            size=version.size,
+            self._serialize(),
+            size=newest_version.size,
             downloads=self.get_download_count(),
             checkout=self.checkout._id if self.checkout else None,
-            version=version.identifier if self.versions.exists() else None,
-            contentType=version.content_type if self.versions.exists() else None,
-            modified=version.date_modified.isoformat() if version.date_modified else None,
-            created=self.versions.first().date_modified.isoformat() if self.versions.first().date_modified else None,
+            version=newest_version.identifier if newest_version else None,
+            contentType=newest_version.content_type if newest_version else None,
+            modified=newest_version.date_modified.isoformat() if newest_version.date_modified else None,
+            created=self.versions.all().first().date_modified.isoformat() if self.versions.all().first().date_modified else None,
         )
 
+    def restore(self, recursive=True, parent=None, save=True, deleted_on=None):
+        raise UnableToRestore('You cannot restore something that is not deleted.')
+
 
-class Folder(FileNode):
-    is_file = False
+class Folder(models.Model):
+    _is_file = False
+
+    class Meta:
+        abstract = True
 
     @property
-    def children(self):
-        """Finds all Filenodes that view self as a parent
-        :returns: A GenWrapper for all children
-        :rtype: GenWrapper<MongoQuerySet<cls>>
-        """
-        return FileNode.find(Q('parent_id', 'eq', self.id))
+    def kind(self):
+        return 'folder'
 
-    def delete(self, recurse=True, user=None, parent=None):
-        trashed = self._create_trashed(user=user, parent=parent)
-        if recurse:
-            for child in self.children:
-                child.delete(user=user, parent=trashed)
-        self._repoint_guids(trashed)
-        StoredFileNode.remove_one(self.stored_object)
-        return trashed
+    @property
+    def children(self):
+        return self._children.all()
```

Comment:
Possibly the cause of https://github.com/CenterForOpenScience/osf.io/pull/6965#issuecomment-288499510